torchmetrics/metric.py

```diff
@@ -224,10 +229,10 @@ def _sync_dist(self, dist_sync_fn: Callable = gather_all_tensors, process_group:
             reduced = reduction_fn(output_dict[attr]) if reduction_fn is not None else output_dict[attr]
             setattr(self, attr, reduced)
 
-    def _wrap_update(self, update):
+    def _wrap_update(self, update: Callable) -> Callable:
 
         @functools.wraps(update)
-        def wrapped_func(*args, **kwargs):
+        def wrapped_func(*args: Tensor, **kwargs: Any) -> Optional[Any]:
```

Comment:
```suggestion
        def wrapped_func(*args: Any, **kwargs: Any) -> Optional[Any]:
```

This probably has to match the signature of `update`